tests/test_report_parsing.py

```diff
@@ -46,7 +71,22 @@ def test_sarif_parser() -> None:
     reports = os.listdir(reports_base_dir)
     for report in reports:
         report_tool = report.removesuffix(".sarif")
+        active_issues = get_issue_codes_for_tool(report_tool)
         report_path = Path(reports_base_dir, report).as_posix()
+
         with parse_single_artifact(report_path, report_tool) as result:
+            missing_ids = []
             assert result
             assert result["issues"] is not None
+            for issue in result["issues"]:
+                issue_code = issue["issue_code"]
+                if issue_code not in active_issues:
+                    missing_ids.append(issue_code)
```

Comment:
```python3
missing_ids = set(issue["issue_code"] for issue in result["issues"]) - set(active_issues)
```

I believe this can be written in a much cleaner way like this.

Reply:
This will be much faster as well.

Reply:
though you can use set comprehension
```suggestion
            missing_ids = {issue["issue_code"] for issue in result["issues"]} - active_issues
```

Reply:
Right. Missed it. Was generating a issue code to id mapping first. Didn't look at it again when I changed the dictionary to set. 

Good catch!